src/views/ProtestPage.js

```diff
@@ -0,0 +1,332 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components/macro';
+import { useHistory, useParams } from 'react-router-dom';
+import { fetchProtest, updateProtest } from '../api';
+import { Map, TileLayer, Marker } from 'react-leaflet';
+import { ProtestForm } from '../components';
+import { Switch, Route } from 'react-router-dom';
+import SocialButton, { Button } from '../components/Button/SocialButton';
+
+const mobile = `@media (max-width: 500px)`;
+
+function useFetchProtest() {
+  const [protest, setProtest] = useState(null);
+  const { id } = useParams();
+
+  useEffect(() => {
+    async function _fetchProtest(id) {
+      const result = await fetchProtest(id);
+      if (result) {
+        setProtest(result);
+      } else {
+        // TODO: handle 404
+      }
+    }
+
+    _fetchProtest(id);
+  }, [id]);
+
+  return protest;
+}
+
+function ProtestPageContent({ protest }) {
+  const history = useHistory();
+
+  const { coordinates, whatsAppLink, telegramLink, displayName, streetAddress, notes } = protest;
+
+  return (
+    <ProtestPageContainer>
+      <MapWrapper center={{ lat: coordinates.latitude, lng: coordinates.longitude }} zoom={14}>
+        <TileLayer
+          attribution='&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        />
+        <Marker position={{ lat: coordinates.latitude, lng: coordinates.longitude }}></Marker>
+      </MapWrapper>
+
+      <ProtestContainer>
+        <Info>
+          <ProfilePic src="/protest-profile-pic.png" alt="Protester with flag getting sprayed" />
+          <Details>
+            <Left>
+              <Title>{displayName}</Title>
+              <Location>
+                <FlagIcon src="/icons/blue-flag.svg" alt="flag icon" />
+                {streetAddress}
+              </Location>
+              <Notes>{notes}</Notes>
+            </Left>
+            <EditButton onClick={() => history.push('edit')}>עריכה</EditButton>
+          </Details>
+        </Info>
+
+        <DatesAndSocial>
+          {/* Dates */}
+          <SectionContainer>
+            <SectionTitle>
+              <img src="/icons/clock.svg" alt="clock icon" />
+              מועדי הפגנה קרובים
+            </SectionTitle>
+
+            <Dates>
+              <Date>
+                <BoldDateText>9.10.2020</BoldDateText>
```

Comment:
It's better to call the component by it's purpose and not by it's style imo. For example, this could be `NumericDate`, and the other could be `DayAndTime`.

Reply:
👍

Reply:
I'm actually looking at these more like classes than like components. And if this were a class i'd probably call it something with bold.

Reply:
Even with classes, I'd call them by what the element is. So `protest-card__date`, even if it's bold - that's for the style decleration itself. 
If we'll start adding style values to our class/styled components names, we can end up with `16pxArialGreenDate` 😛